### PR TITLE
Budget App

### PR DESCRIPTION
**Pull request by muhammedb8826**
In this project, I have implemented and worked on the following:

- Interactions
Splash screen
Created A simple page with the name of my app name and links to the sign-up and log-in pages.

The user should be able to register in the app with full name, email, and password (all mandatory).
The user can log into the app using email and password.
If the user is not logged in, they can't access pages that require them to be logged in (all the pages described below).

- Home page (categories page)

When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
A button "add a new category" at the bottom brings the user to the page to create a new category.

- Transactions page

The list of transactions is presented for a given category, ordered by the most recent.
At the top of the page, the user can see the total amount for the category (the sum of all of the amounts of the transactions in that category).
A button "add a new transaction" at the bottom brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page

The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page

The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: app/views/deals/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<h1 class="new-form">New transaction</h1>
+
+<%= render "form", deal: @deal %>
+
+<br>
+
+<div class="back">
+  <%= link_to "Back to transactions", category_path(id: @category.id) %>
+</div>
```
Comment (codepantha) [inline app/views/deals/new.html.erb]: - [ ] Please place the "Back" button (<) at the top navigation of the page just as in the Behance design shown below 👇 
### Your Design
image
### Design Guideline
image

File: app/views/categories/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<h1 class="new-form">New category</h1>
+
+<%= render "form", category: @category %>
+
+<br>
+
+<div class="back">
+  <%= link_to "Back to categories", categories_path %>
+</div>
```
Comment (codepantha) [inline app/views/categories/new.html.erb]: - [ ] Please place the "Back" button (<) at the top navigation of the page just as in the Behance design shown below 👇 
### Your Design
image
### Design Guideline
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @muhammedb8826 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯
✔️ Descriptive PR
✔️ Video presentation audio is clear 📌 
✔️ App working as expected 📌 
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

- [ ] Your deployed app fails to show the sign up page. Kindly see the error shown below and ensure that it is properly deployed 👍 
image
- [ ] Please write integration tests for all the most important components of the RoR application 👍 
- [ ] For your video presentation, please ensure that the  video contains a highlight about a piece of the code that you find interesting or are proud of making 😎 
image



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 3]
```diff
@@ -1 +1,182 @@
...
+- Assign a transaction to a category
+
+<p align="right"><a href="#readme-top">👆</a></p>
+
+<!-- Getting Started -->
+
+## 🚀 Live Demo
+
+[Click here to see the page](https://budget-app-xoqr.onrender.com/)
+
```
Comment (elisha2kyakpo1) [inline README.md:81]: - [ ] As requested by the previous reviewer, kindly fix the live link sign-up/sign in URL. I got the error in the image below.

live-app

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @muhammedb8826, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

👏 You did well to fix the back button position as requested by the previous reviewer.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by muhammedb8826** [comment]
Hello @elisha2kyakpo1, Thank you for the feedback and your time.

I am able to fix the deployment error

here is the screenshot for the working application on the render

working project


details page

**Pull request merged**